### PR TITLE
[PATCH v2] linux-gen: fix process mode issues

### PR DESCRIPTION
**Pull request by matiaselo**
V2:
- Rebase

V3:
- Added testing and Travis support

V4:
- Put only the modified option in `process-mode.conf` (Bill)

### CODE REVIEW

File: config/odp-linux-generic.conf [hunk 1]
```diff
@@ -33,6 +33,10 @@ shm: {
 	# When using process mode threads, this value should be set to 0
 	# because the current implementation won't work properly otherwise.
 	num_cached_hp = 0
+
+	# Allocate internal shared memory using a single virtual address space.
+	# Set to 1 to enable using process mode.
+	single_va = 0
```
Comment (Bill-Fischofer-Linaro) [inline config/odp-linux-generic.conf]: This seems to be duplicate of PR #706. Just temporary?

Given the scope of these changes, we really need to test with `single_va = 1` as well as the default 0 here.

**Comment by Bill-Fischofer-Linaro** [comment]
Travis has issues with IPC test. Needs investigation.

File: config/odp-linux-generic.conf (hunk 1 above)
Comment (matiaselo) [reply config/odp-linux-generic.conf]: >This seems to be duplicate of PR #706. Just temporary?

Yes, the first three commits are temporarily copied from PR #706 as this patch set builds on top of that PR.

**Comment by matiaselo** [comment]
>Travis has issues with IPC test. Needs investigation.

This issue seems to happen quite randomly as my private Travis run passed. I suspect that the problem may be in here:
```
+		uint32_t flags = ODP_SHM_EXPORT;
+
+		if (odp_global_data.shm_single_va)
+			flags |= ODP_SHM_SINGLE_VA;
+
 		pkt_priv(pktio_entry)->type = PKTIO_TYPE_IPC_MASTER;
 		snprintf(name, sizeof(name), "%s_info", dev);
 		shm = odp_shm_reserve(name, sizeof(struct pktio_info),
-				      ODP_CACHE_LINE_SIZE,
-				      _ODP_ISHM_EXPORT | _ODP_ISHM_LOCK);
+				      ODP_CACHE_LINE_SIZE, flags);

```

The original code was passing internal defines to odp_shm_reserve() and it looked like _ODP_ISHM_LOCK was ignored internally. Taking a more closer look it seems _ODP_ISHM_LOCK maps to ODP_SHM_SINGLE_VA. @muvarov is this a correct interpretation?

**Comment by muvarov** [comment]
Interesting that normal exit was printed but exit status is not 0 for ipc under coverage. I will re run this test.
```
odp_packet_io.c:521:odp_pktio_start():interface: ipc:23935:ipktio, input queues: 1, output queues: 1
odp_packet_io.c:573:odp_pktio_stop():interface: ipc:23935:ipktio
odp_packet_io.c:410:odp_pktio_close():interface: ipc:23935:ipktio
normal exit of 2 application
!!!First stage  FAILED 255!!!
FAIL pktio_ipc/pktio_ipc_run.sh (exit status: 255)
```

**Comment by matiaselo** [comment]
> Interesting that normal exit was printed but exit status is not 0 for ipc under coverage. I will re run this test.

@muvarov That's odd, I can't repeat this on my system and the coverage test seems to be now passing. This PR doesn't really change anything in IPC, so this failure was probably unrelated (unless the failure happens in process mode).

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**